File: app/src/main/java/com/falcon/restaurants/screens/restaurant/RestaurantViewModelFactory.java
package com.falcon.restaurants.screens.restaurant;

import android.app.Application;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class RestaurantViewModelFactory implements ViewModelProvider.Factory {

    Application application ;
    FetchRestaurantsUseCase fetchRestaurantsUseCase;

    public RestaurantViewModelFactory(Application application, FetchRestaurantsUseCase fetchRestaurantsUseCase) {
        this.application = application;
        this.fetchRestaurantsUseCase = fetchRestaurantsUseCase;
    }

    @androidx.annotation.NonNull
    @Override
    public <T extends ViewModel> T create(@androidx.annotation.NonNull Class<T> modelClass) {
        return (T) new RestaurantViewModel(application, fetchRestaurantsUseCase);
    }
}

File: app/src/main/java/com/falcon/restaurants/screens/restaurant/RestaurantViewModelFactory.kt
package com.falcon.restaurants.screens.restaurant

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class RestaurantViewModelFactory(
    var application: Application,
    var fetchRestaurantsUseCase: FetchRestaurantsUseCase
) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return RestaurantViewModel(application, fetchRestaurantsUseCase) as T
    }
}